### PR TITLE
Minimal bounding region in spectral extraction

### PR DESCRIPTION
**Pull request by ibusko**
As per #746 

This PR must be reviewed  and merged only **AFTER** PR #753 is merged.

This PR contains changes that generate conflicts with PR #753, thus it was developed on top of the 753 branch, and not on top of master. Thus the need for the ordered merging  (the relevant changes corresponding only to this PR are in the 2nd and later commits).

The CI doc build error seems to be coming from a different module (docs/fitting.rst).

### CODE REVIEW

**Review by rosteen — APPROVED** [review]
Looks good to me (other than the minor conflicts to be resolved with the spectral_slab PR).

**Review by javerbukh — APPROVED** [review]
Looks good!

**Comment by eteq** [comment]
#753 is in so I think this can be rebased, @ibusko ?

**Comment by eteq** [comment]
(I rebased it myself, because I realized I can just merge it as it stands if I do that and the tests pass)

**Comment by eteq** [comment]
Note: I'm realizing I'm not sure this is quite what #746 is about - see https://github.com/astropy/specutils/issues/746#issuecomment-770087237 there.  If @rosteen and/or @ibusko agree with my comment there, then I think we want to change the name of this PR's function to `extract_bounding_spectral_region`.  Alternatively we could leave it as-is and plan on adding on the "spatial" part of this function later, as it would then let you extract the minimal bounding region in *any* of the axes.  But if we go that route we should definitely make an issue recording it before merging this.

**Comment by ibusko** [comment]
@eteq yes, the `minimal_subcube` function applies to both spatial and spectral regions. I haven't realized this before, I think I was driven by the description in #746 , which seems to apply to the spectral part only. Besides, the concept as applied to the spatial part only makes sense when masks are supported. In the spectral domain, we have `SpectralRegion` 's that may be thought as a kind of "mask". So, perhaps your first suggestion (change the function name) is the way to go. For now.

**Comment by ibusko** [comment]
I renamed the function, and created issue #774 to record the need to implement the spatial handling.

File: specutils/manipulation/extract_spectral_region.py
Comment (eteq) [inline specutils/manipulation/extract_spectral_region.py:222]: This should fix the docs failure in the PR
```suggestion
        sub-regions in the input ``region`` instance.
```

**Comment by eteq** [comment]
In https://github.com/astropy/specutils/pull/755#pullrequestreview-580496961 I fixed what I think was the reason for the test failures.  Will merge this if the tests pass

**Review by eteq — APPROVED** [review]

**Pull request merged**
